### DISCUSSION TITLE
Running lighthouse programmatically in docker

### DISCUSSION THREAD

**Discussion by gbajpeyee** [GoogleChrome/lighthouse#11898] [Q&A]
I am running lighthouse programmatically in inside docker container. While running inside container I get the following error.
Do I need to have chrome image installed in my container to fix it or there is other way to fix it ? 
```
const lighthouse = require('lighthouse');
const chromeLauncher = require('chrome-launcher');

async function getLighthouseResults(url) {
    try {
        const chrome = await chromeLauncher.launch({ chromeFlags: ['--headless'] });
        const options = { logLevel: 'info', output: 'html', onlyCategories: ['performance'], port: chrome.port };
        const runnerResult = await lighthouse(url, options);
        // `.report` is the HTML report as a string
        const reportHtml = runnerResult.report;
        // `.lhr` is the Lighthouse Result as a JS object
        logger.info('Report is done for', runnerResult.lhr.finalUrl);
        logger.info('Performance score was', runnerResult.lhr.categories.performance.score * 100);
        await chrome.kill();
        return reportHtml;
    } catch (error) {
        logger.error('Error encountered during lighthouse', url, error);
        return error;
    }
}

getLighthouseResults('https://www.google.com/');
```
> Error encountered during lighthouse http://test.com ChromePathNotSetError\n at new LauncherError (/opt/att/app/node_modules/chrome-launcher/dist/utils.js:37:22)\n at new ChromePathNotSetError (/opt/att/app/node_modules/chrome-launcher/dist/utils.js:44:9)\n at Object.linux (/o

**Comment by patrickhulce** [comment]
> Do I need to have chrome image installed in my container to fix it or there is other way to fix it ?

Yes Chrome needs to be installed in your container. See some [official Lighthouse docker examples](https://github.com/GoogleChrome/lighthouse-ci/blob/7f46602f1e2d86effcda994ec951664982e3dc8a/docs/recipes/docker-client/Dockerfile#L3-L11) on how to get it done.

**Reply by gbajpeyee** [reply]
Thank you !!
